network/mode/bridge/bridge.go

```diff
@@ -148,15 +115,18 @@ func New(ctx context.Context, config network.BridgeConfig, manager mgr.NetworkMg
 	return err
 }
 
-func initBridgeDevice(name string) (netlink.Link, error) {
+func initBridgeDevice(name string, ipNet *net.IPNet) (netlink.Link, error) {
 	br, err := netlink.LinkByName(name)
-	if err == nil && br != nil {
-		return br, nil
+	if err == nil && br != nil { // we need delete old link
+		err := netlink.LinkDel(br)
```

Comment:
Add a condition, if the exist bridge's ip is same with config bip, it no need to delete old bridge. And check the veth-pair is attached on the exist bridge, this case can't delete bridge device.